### ISSUE TITLE
🐛 Biome crashed while open astro file

### ISSUE THREAD

**Issue by shaddollxz** [biomejs/biome#7837]
### Environment information

```block
CLI:
  Version:                      2.3.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.18.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.20

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

while open an astro file, biome crashed, i get this error in `Biome LSP` output

```
2025-10-24 18:06:08.046 [info] Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /Users/runner/work/biome/biome/crates/biome_parser/src/lib.rs:888:13
Thread Name: tokio-runtime-worker
Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
```

there is my code and config

```astro
---
import { Counter } from '@/components/counter'
import PageLayout from '@/components/page-layout/index.astro'

// Welcome to Astro! Wondering what to do next? Check out the Astro documentation at https://docs.astro.build
// Don't want to use any of this? Delete everything in this file, the `assets`, `components`, and `layouts` directories, and start fresh.
---
<PageLayout>

  <main class="flex h-screen w-screen flex-col items-center justify-center">
    <Counter client:load>

      <span>------</span>
    </Counter>
  </main>
</PageLayout>

```

```jsonc
// biome.jsonc
{
 "html": {
    "experimentalFullSupportEnabled": true
  }
}
```

### Expected result

not crashed and format my astro code

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by shaddollxz** [comment]
and i run biome by bun to format or lint for this file, i got those errors:

```sh
❯ bunx biome lint apps/web/src/pages/index.astro --fix  
apps/web/src/pages/index.astro:14:13 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unexpected value or character.
  
    12 │     <Counter client:load>
    13 │ 
  > 14 │       <span>---</span>
       │             ^^^
    15 │     </Counter>
    16 │   </main>
  
  ℹ Expected one of:
  
  - element
  - text
  

Checked 1 file in 28ms. Fixed 1 file.
Found 1 error.
lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

```sh
❯ bunx biome format apps/web/src/pages/index.astro --fix
Formatted 0 files in 3ms. No fixes applied.
format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
  
  ℹ Check your biome.json or biome.jsonc to ensure the paths are not ignored by the configuration.
  
  ℹ These paths were provided but ignored:
  
  - apps/web/src/pages/index.astro
```

here is my full config

biome.jsonc in monorepo root:
```jsonc
{
  "$schema": "https://biomejs.dev/schemas/2.3.0/schema.json",
  "root": true,
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true,
    "defaultBranch": "main"
  },
  "files": {
    "ignoreUnknown": true
  },
  "assist": {
    "enabled": true,
    "actions": {
      "source": {
        "organizeImports": "on",
        "useSortedProperties": "off",
        "useSortedAttributes": "off"
      }
    }
  },
  "formatter": {
    "enabled": true,
    "useEditorconfig": true,
    "formatWithErrors": false,
    "indentStyle": "space",
    "indentWidth": 2,
    "lineEnding": "lf",
    "attributePosition": "auto",
    "bracketSpacing": true
  },
  "linter": {
    "enabled": true,
    "domains": {
      "react": "recommended"
    },
    "rules": {
      "recommended": true,
      "nursery": {
        "useSortedClasses": "error",
        "useVueDefineMacrosOrder": "error"
      },
      "suspicious": {
        "noAlert": "error",
        "noConsole": "error",
        "noEmptyBlockStatements": "warn",
        "noMisplacedAssertion": "error",
        "noSkippedTests": "error",
        "useAwait": "error"
      },
      "a11y": {
        "useKeyWithClickEvents": "off",
        "useButtonType": "off"
      },
      "complexity": {
        "noForEach": "off",
        "noUselessStringConcat": "error",
        "noUselessUndefinedInitialization": "error",
        "noVoid": "error",
        "useDateNow": "error",
        "useSimplifiedLogicExpression": "error"
      },
      "correctness": {
        "useExhaustiveDependencies": "warn",
        "noUnusedFunctionParameters": "error",
        "noUnusedImports": "error",
        "noUnusedPrivateClassMembers": "error",
        "noUnusedVariables": "error"
      },
      "style": {
        "noNamespace": "error",
        "noNegationElse": "error",
        "noYodaExpression": "error",
        "noNonNullAssertion": "off",
        "useConsistentArrayType": {
          "level": "error",
          "options": {
            "syntax": "generic"
          }
        },
        "useConsistentBuiltinInstantiation": "error",
        "useFilenamingConvention": {
          "level": "error",
          "options": {
            "filenameCases": ["kebab-case"]
          }
        }
      }
    }
  },
  "javascript": {
    "jsxRuntime": "transparent",
    "formatter": {
      "jsxQuoteStyle": "double",
      "quoteProperties": "asNeeded",
      "trailingCommas": "all",
      "semicolons": "asNeeded",
      "arrowParentheses": "always",
      "bracketSameLine": false,
      "quoteStyle": "single",
      "attributePosition": "auto",
      "bracketSpacing": true
    }
  },
  "json": {
    "formatter": {
      "trailingCommas": "none"
    }
  },
  "css": {
    "parser": { "tailwindDirectives": true },
    "formatter": {
      "quoteStyle": "single"
    }
  },
  "html": {
    "experimentalFullSupportEnabled": true
  }
}
```

biome.jsonc in workspace:

```jsonc
{
  "$schema": "https://biomejs.dev/schemas/2.3.0/schema.json",
  "root": false,
  "extends": ["./../../biome.jsonc"],
  "overrides": [
    {
      "includes": ["**/*.svelte", "**/*.astro", "**/*.vue"],
      "linter": {
        "rules": {
          "correctness": {
            "noUnusedVariables": "off",
            "noUnusedImports": "off"
          }
        }
      }
    }
  ]
}

```

**Comment by ematipico** [comment]
You must enable `html.formatter.enabled`

**Comment by ematipico** [comment]
[Reproduction](https://biomejs.dev/playground/?code=LQAtAC0ACgBpAG0AcABvAHIAdAAgAHsAIABDAG8AdQBuAHQAZQByACAAfQAgAGYAcgBvAG0AIAAnAEAALwBjAG8AbQBwAG8AbgBlAG4AdABzAC8AYwBvAHUAbgB0AGUAcgAnAAoAaQBtAHAAbwByAHQAIABQAGEAZwBlAEwAYQB5AG8AdQB0ACAAZgByAG8AbQAgACcAQAAvAGMAbwBtAHAAbwBuAGUAbgB0AHMALwBwAGEAZwBlAC0AbABhAHkAbwB1AHQALwBpAG4AZABlAHgALgBhAHMAdAByAG8AJwAKAAoALwAvACAAVwBlAGwAYwBvAG0AZQAgAHQAbwAgAEEAcwB0AHIAbwAhACAAVwBvAG4AZABlAHIAaQBuAGcAIAB3AGgAYQB0ACAAdABvACAAZABvACAAbgBlAHgAdAA%2FACAAQwBoAGUAYwBrACAAbwB1AHQAIAB0AGgAZQAgAEEAcwB0AHIAbwAgAGQAbwBjAHUAbQBlAG4AdABhAHQAaQBvAG4AIABhAHQAIABoAHQAdABwAHMAOgAvAC8AZABvAGMAcwAuAGEAcwB0AHIAbwAuAGIAdQBpAGwAZAAKAC8ALwAgAEQAbwBuACcAdAAgAHcAYQBuAHQAIAB0AG8AIAB1AHMAZQAgAGEAbgB5ACAAbwBmACAAdABoAGkAcwA%2FACAARABlAGwAZQB0AGUAIABlAHYAZQByAHkAdABoAGkAbgBnACAAaQBuACAAdABoAGkAcwAgAGYAaQBsAGUALAAgAHQAaABlACAAYABhAHMAcwBlAHQAcwBgACwAIABgAGMAbwBtAHAAbwBuAGUAbgB0AHMAYAAsACAAYQBuAGQAIABgAGwAYQB5AG8AdQB0AHMAYAAgAGQAaQByAGUAYwB0AG8AcgBpAGUAcwAsACAAYQBuAGQAIABzAHQAYQByAHQAIABmAHIAZQBzAGgALgAKAC0ALQAtAAoAPABQAGEAZwBlAEwAYQB5AG8AdQB0AD4ACgAKACAAIAA8AG0AYQBpAG4AIABjAGwAYQBzAHMAPQAiAGYAbABlAHgAIABoAC0AcwBjAHIAZQBlAG4AIAB3AC0AcwBjAHIAZQBlAG4AIABmAGwAZQB4AC0AYwBvAGwAIABpAHQAZQBtAHMALQBjAGUAbgB0AGUAcgAgAGoAdQBzAHQAaQBmAHkALQBjAGUAbgB0AGUAcgAiAD4ACgAgACAAIAAgADwAQwBvAHUAbgB0AGUAcgAgAGMAbABpAGUAbgB0ADoAbABvAGEAZAA%2BAAoACgAgACAAIAAgACAAIAA8AHMAcABhAG4APgAtAC0ALQAtAC0ALQA8AC8AcwBwAGEAbgA%2BAAoAIAAgACAAIAA8AC8AQwBvAHUAbgB0AGUAcgA%2BAAoAIAAgADwALwBtAGEAaQBuAD4ACgA8AC8AUABhAGcAZQBMAGEAeQBvAHUAdAA%2BAA%3D%3D&language=astro)

@shaddollxz, please, next time, provide a reproduction

**Comment by shaddollxz** [comment]
> [Reproduction](https://biomejs.dev/playground/?code=LQAtAC0ACgBpAG0AcABvAHIAdAAgAHsAIABDAG8AdQBuAHQAZQByACAAfQAgAGYAcgBvAG0AIAAnAEAALwBjAG8AbQBwAG8AbgBlAG4AdABzAC8AYwBvAHUAbgB0AGUAcgAnAAoAaQBtAHAAbwByAHQAIABQAGEAZwBlAEwAYQB5AG8AdQB0ACAAZgByAG8AbQAgACcAQAAvAGMAbwBtAHAAbwBuAGUAbgB0AHMALwBwAGEAZwBlAC0AbABhAHkAbwB1AHQALwBpAG4AZABlAHgALgBhAHMAdAByAG8AJwAKAAoALwAvACAAVwBlAGwAYwBvAG0AZQAgAHQAbwAgAEEAcwB0AHIAbwAhACAAVwBvAG4AZABlAHIAaQBuAGcAIAB3AGgAYQB0ACAAdABvACAAZABvACAAbgBlAHgAdAA%2FACAAQwBoAGUAYwBrACAAbwB1AHQAIAB0AGgAZQAgAEEAcwB0AHIAbwAgAGQAbwBjAHUAbQBlAG4AdABhAHQAaQBvAG4AIABhAHQAIABoAHQAdABwAHMAOgAvAC8AZABvAGMAcwAuAGEAcwB0AHIAbwAuAGIAdQBpAGwAZAAKAC8ALwAgAEQAbwBuACcAdAAgAHcAYQBuAHQAIAB0AG8AIAB1AHMAZQAgAGEAbgB5ACAAbwBmACAAdABoAGkAcwA%2FACAARABlAGwAZQB0AGUAIABlAHYAZQByAHkAdABoAGkAbgBnACAAaQBuACAAdABoAGkAcwAgAGYAaQBsAGUALAAgAHQAaABlACAAYABhAHMAcwBlAHQAcwBgACwAIABgAGMAbwBtAHAAbwBuAGUAbgB0AHMAYAAsACAAYQBuAGQAIABgAGwAYQB5AG8AdQB0AHMAYAAgAGQAaQByAGUAYwB0AG8AcgBpAGUAcwAsACAAYQBuAGQAIABzAHQAYQByAHQAIABmAHIAZQBzAGgALgAKAC0ALQAtAAoAPABQAGEAZwBlAEwAYQB5AG8AdQB0AD4ACgAKACAAIAA8AG0AYQBpAG4AIABjAGwAYQBzAHMAPQAiAGYAbABlAHgAIABoAC0AcwBjAHIAZQBlAG4AIAB3AC0AcwBjAHIAZQBlAG4AIABmAGwAZQB4AC0AYwBvAGwAIABpAHQAZQBtAHMALQBjAGUAbgB0AGUAcgAgAGoAdQBzAHQAaQBmAHkALQBjAGUAbgB0AGUAcgAiAD4ACgAgACAAIAAgADwAQwBvAHUAbgB0AGUAcgAgAGMAbABpAGUAbgB0ADoAbABvAGEAZAA%2BAAoACgAgACAAIAAgACAAIAA8AHMAcABhAG4APgAtAC0ALQAtAC0ALQA8AC8AcwBwAGEAbgA%2BAAoAIAAgACAAIAA8AC8AQwBvAHUAbgB0AGUAcgA%2BAAoAIAAgADwALwBtAGEAaQBuAD4ACgA8AC8AUABhAGcAZQBMAGEAeQBvAHUAdAA%2BAA%3D%3D&language=astro)
> 
> [@shaddollxz](https://github.com/shaddollxz), please, next time, provide a reproduction

Here is the repository with my minimal reproduction code. The reproduction steps are already included in the README: https://github.com/shaddollxz/biome-astro

**Comment by nhedger** [comment]
Here's a similar reproduction: https://github.com/nhedger/biome-repro-vue

This happens on both Astro and Vue files, at least

**Comment by roonie007** [comment]
> You must enable `html.formatter.enabled`

Same error even with HTML formater enabled for a Vue3 project

```
2025-10-25 00:23:41.782 [info] Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /__w/biome/biome/crates/biome_parser/src/lib.rs:888:13
Thread Name: tokio-runtime-worker
Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
```

```jsonc
{
  // ---- HTML / Vue SFC template formatting (Biome 2.3)
  "html": {
    // Enables full SFC template formatting support
    "experimentalFullSupportEnabled": true,
    "formatter": {
      "enabled": true,
      "indentScriptAndStyle": false,
      "selfCloseVoidElements": "always"
    }
  },
}
```

Also I want to note that the `biome check` works perfectly, it's just the VsCode extention that is throwing an error.

<img width="652" height="1125" alt="Image" src="https://github.com/user-attachments/assets/6fb042d8-876a-44a3-913b-edbd20f3e806" />

**Comment by roonie007** [comment]
Here is some more logs, this is provided as soon as I open a .vue file


```
└─┐biome_lsp::handlers::text_document::did_open{text_document_uri=file:///MY_PATH/app/components/ui/spinner/spinner.vue, text_document_language_id=vue}
  └─┐biome_service::file_handlers::try_from_language_id{language_id="vue"}
  ┌─┘
  └─┐biome_service::workspace::server::open_file_internal{reason=ClientRequest, path=/MY_PATH/app/components/ui/spinner/spinner.vue}
    ├─ DEBUG biome_service::file_handlers::vue Parsing file with language JsFileSource { language: TypeScript { definition_file: false }, variant: Standard, module_kind: Module, version: ES2022, embedding_kind: Vue }
    ├─ ERROR biome_cli::panic Biome encountered an unexpected error
    │  
    │  This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.
    │  
    │  When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:
    │  
    │  Source Location: /__w/biome/biome/crates/biome_parser/src/lib.rs:888:13
    │  Thread Name: tokio-runtime-worker
    │  Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
    │  
  ┌─┘
┌─┘
┘
INFO biome_cli::commands::daemon Received shutdown signal
```

**Comment by axel-rock** [comment]
Similar issue on my Svelte project.

Happens in a few files.

Here is the shortest one:

```
<script lang="ts" module>
  export const selectedMessages = new SvelteSet<UIMessage>();
</script>

<script lang="ts">
  import type { UIMessage } from "ai";
  import Message from "$lib/components/chat/Message.svelte";
  import { SvelteSet } from "svelte/reactivity";
  import type { Attachment } from "svelte/attachments";

  const {
    messages,
  }: {
    messages: UIMessage[];
  } = $props();

  let initialScroll = $state(true);

  const autoScroll: Attachment = (element) => {
    messages.length &&
      element.lastElementChild?.scrollIntoView({
        behavior: initialScroll ? "smooth" : "instant",
        block: "start",
      });
    initialScroll = true;
  };
</script>

<div class="messages grow" {@attach autoScroll}>
  {#each messages as message (message.id)}
    <Message {message} />
  {/each}
</div>

<style>
  .messages {
    display: flex;
    flex-direction: column;
    scroll-snap-type: y mandatory;
    scroll-behavior: smooth;
    gap: 0.5rem;
    padding-bottom: 2rem;
  }
</style>

```



```
src/lib/components/chat/Messages.svelte internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: The parser is no longer progressing. Stuck at '{' L_CURLY:704..705

  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
```

I managed to find this log in Zed. I assume it's also what happens in VSCode.

**Comment by dyc3** [comment]
@axel-rock at a glance, that seems like a separate issue in the html parser. would you mind filing a new issue with a minimal repro?

**Comment by qyurila** [comment]
Actually it seems just opening any Astro file causes this.

## Reproduction Step

1. Start a new project (no matter if npm is used instead):

```sh
bun add -D @biomejs/biome
bunx biome init
```

2. Enable the experimental option in `biome.jsonc`:

```json
	"html": {
		"experimentalFullSupportEnabled": true
	}
```

3. Create any valid `.astro` file:

```sh
touch index.astro && echo "<div></div>" > ./index.astro
```

4. Open the above file on VSCode with Biome extension enabled.

## Output

```
2025-10-27 10:45:30.158 [info] Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: C:\actions-runner\biome\biome\crates\biome_parser\src\lib.rs:894:13
Thread Name: tokio-runtime-worker
Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
```

## Spec

- Biome extension version: v3.3.0
- `bunx biome rage` output:
<details>
<summary>Click to open</summary>

```
CLI:
  Version:                      2.3.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.20.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```
</details>

**Comment by ematipico** [comment]
@qyurila @roonie007 

The error message is different from OP, it seems?

Well, we'll track everything here 🤷

**Comment by qyurila** [comment]
> [@qyurila](https://github.com/qyurila) [@roonie007](https://github.com/roonie007)
> 
> The error message is different from OP, it seems?
> 
> Well, we'll track everything here 🤷

They're literally identical tho:

```
Source Location: (...)/biome/biome/crates/biome_parser/src/lib.rs:888:13
Thread Name: tokio-runtime-worker
Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
```

The only difference is the error position in mine being `894:13` but that's caused by Biome version differences (v2.3.1 vs. v2.3.0) if that's what you're talking about.

**Comment by roonie007** [comment]
I can confirm it's the same message

> could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage

**Comment by ematipico** [comment]
Yeah sorry. The original issue was about Astro parsing/crash, and then we added Vue files, which might have different parsing branch.

While the crash hints at the same message, the cause might be different.

So for my mental sanity, I was hoping to have two separate issues to track.

**Comment by adarshmadrecha** [comment]
Using `v2.3.1` the issue is still there for `vue` js project. 

```bash
2025-10-29 13:57:48.076 [info] Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: C:\actions-runner\biome\biome\crates\biome_parser\src\lib.rs:894:13
Thread Name: tokio-runtime-worker
Message: could not downcast root node to language biome_html_syntax::syntax_node::HtmlLanguage
```


<details><summary>Biome Config</summary>
<p>

```json
{
  "$schema": "https://biomejs.dev/schemas/2.3.1/schema.json",
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "files": {
    "ignoreUnknown": false
  },
  "formatter": {
    "enabled": true,
    "formatWithErrors": false,
    "indentStyle": "space",
    "indentWidth": 2
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "correctness": {
        "noUnusedVariables": "off"
      },
      "suspicious": {
        "noTsIgnore": "off"
      },
      "style": {
        "noNonNullAssertion": "off"
      }
    }
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single",
      "semicolons": "asNeeded",
      "lineWidth": 120
    }
  },
  "html": {
    "formatter": {
      "enabled": true
    },
    "experimentalFullSupportEnabled": true
  },
  "css": {
    "parser": {
      "tailwindDirectives": true
    }
  },
  "assist": {
    "enabled": true,
    "actions": {
      "source": {
        "organizeImports": {
          "level": "off",
          "options": {
            "identifierOrder": "lexicographic"
          }
        }
      }
    }
  }
}

```

</p>
</details>

**Comment by ematipico** [comment]
@adarshmadrecha you aren't using the latest. Please update

**Comment by patrikengborg** [comment]
I am using the latest (2.3.2), and I have the exact same error in my SvelteKit project.

**Comment by qyurila** [comment]
@patrikengborg Just to be clear, please check if the latest binary is actually used by the extension. Does the Biome status bar item show `2.3.2` (If you're using VSCode)?

Just mentioning because it didn't until I manually set `"biome.lsp.bin"` option in VSCode config. After that the issue has been fixed for me.

If the issue still persists you may want to open another issue and bring another reproduction, because the title says Astro and the issue seems fixed for Astro at least.

**Comment by patrikengborg** [comment]
@qyurila Worked! Thanks.

**Comment by wtto00** [comment]
> [Reproduction](https://biomejs.dev/playground/?code=LQAtAC0ACgBpAG0AcABvAHIAdAAgAHsAIABDAG8AdQBuAHQAZQByACAAfQAgAGYAcgBvAG0AIAAnAEAALwBjAG8AbQBwAG8AbgBlAG4AdABzAC8AYwBvAHUAbgB0AGUAcgAnAAoAaQBtAHAAbwByAHQAIABQAGEAZwBlAEwAYQB5AG8AdQB0ACAAZgByAG8AbQAgACcAQAAvAGMAbwBtAHAAbwBuAGUAbgB0AHMALwBwAGEAZwBlAC0AbABhAHkAbwB1AHQALwBpAG4AZABlAHgALgBhAHMAdAByAG8AJwAKAAoALwAvACAAVwBlAGwAYwBvAG0AZQAgAHQAbwAgAEEAcwB0AHIAbwAhACAAVwBvAG4AZABlAHIAaQBuAGcAIAB3AGgAYQB0ACAAdABvACAAZABvACAAbgBlAHgAdAA%2FACAAQwBoAGUAYwBrACAAbwB1AHQAIAB0AGgAZQAgAEEAcwB0AHIAbwAgAGQAbwBjAHUAbQBlAG4AdABhAHQAaQBvAG4AIABhAHQAIABoAHQAdABwAHMAOgAvAC8AZABvAGMAcwAuAGEAcwB0AHIAbwAuAGIAdQBpAGwAZAAKAC8ALwAgAEQAbwBuACcAdAAgAHcAYQBuAHQAIAB0AG8AIAB1AHMAZQAgAGEAbgB5ACAAbwBmACAAdABoAGkAcwA%2FACAARABlAGwAZQB0AGUAIABlAHYAZQByAHkAdABoAGkAbgBnACAAaQBuACAAdABoAGkAcwAgAGYAaQBsAGUALAAgAHQAaABlACAAYABhAHMAcwBlAHQAcwBgACwAIABgAGMAbwBtAHAAbwBuAGUAbgB0AHMAYAAsACAAYQBuAGQAIABgAGwAYQB5AG8AdQB0AHMAYAAgAGQAaQByAGUAYwB0AG8AcgBpAGUAcwAsACAAYQBuAGQAIABzAHQAYQByAHQAIABmAHIAZQBzAGgALgAKAC0ALQAtAAoAPABQAGEAZwBlAEwAYQB5AG8AdQB0AD4ACgAKACAAIAA8AG0AYQBpAG4AIABjAGwAYQBzAHMAPQAiAGYAbABlAHgAIABoAC0AcwBjAHIAZQBlAG4AIAB3AC0AcwBjAHIAZQBlAG4AIABmAGwAZQB4AC0AYwBvAGwAIABpAHQAZQBtAHMALQBjAGUAbgB0AGUAcgAgAGoAdQBzAHQAaQBmAHkALQBjAGUAbgB0AGUAcgAiAD4ACgAgACAAIAAgADwAQwBvAHUAbgB0AGUAcgAgAGMAbABpAGUAbgB0ADoAbABvAGEAZAA%2BAAoACgAgACAAIAAgACAAIAA8AHMAcABhAG4APgAtAC0ALQAtAC0ALQA8AC8AcwBwAGEAbgA%2BAAoAIAAgACAAIAA8AC8AQwBvAHUAbgB0AGUAcgA%2BAAoAIAAgADwALwBtAGEAaQBuAD4ACgA8AC8AUABhAGcAZQBMAGEAeQBvAHUAdAA%2BAA%3D%3D&language=astro)
> 
> [@shaddollxz](https://github.com/shaddollxz), please, next time, provide a reproduction

`<span>------</span>` in `astro/vue` still `parse: Unexpected value or character.` in `v2.3.8`!

<details>
<summary>Biome Config</summary>
```json
{
  "$schema": "https://biomejs.dev/schemas/2.3.8/schema.json",
  "files": {
    "ignoreUnknown": true
  },
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "lineWidth": 120,
    "indentWidth": 2,
    "lineEnding": "lf"
  },
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "style": {
        "noNonNullAssertion": "off"
      },
      "correctness": {
        "noUnusedImports": "error",
        "noUnusedVariables": "error"
      }
    },
    "domains": {
      "vue": "recommended",
      "tailwind": "recommended",
      "project": "recommended",
      "solid": "recommended",
      "test": "recommended",
      "qwik": "recommended"
    }
  },
  "javascript": {
    "formatter": {
      "quoteStyle": "single",
      "semicolons": "asNeeded"
    }
  },
  "html": {
    "experimentalFullSupportEnabled": true,
    "formatter": {
      "enabled": true
    }
  },
  "assist": {
    "enabled": true,
    "actions": {
      "source": {
        "organizeImports": "on"
      }
    }
  },
  "css": {
    "parser": {
      "tailwindDirectives": true
    }
  },
  "overrides": [
    {
      "includes": ["**/*.vue"],
      "linter": {
        "rules": {
          "correctness": {
            "noUnusedVariables": "off",
            "noUnusedImports": "off"
          }
        }
      }
    },
    {
      "includes": ["auto-imports.d.ts", "virtual.icons.css", "components.d.ts"],
      "linter": {
        "enabled": false
      }
    },
    {
      "includes": ["index.html"],
      "linter": {
        "rules": {
          "complexity": {
            "noArguments": "off"
          }
        }
      }
    },
    {
      "includes": ["**/*.html"],
      "linter": {
        "rules": {
          "correctness": {
            "noUnusedVariables": "off"
          }
        }
      }
    }
  ]
}
```
</details>
